tools/lib/provision.py

```diff
@@ -318,6 +326,14 @@ def main(options):
         from zerver.lib.str_utils import force_bytes
         from zerver.lib.test_fixtures import is_template_database_current
 
+        # change settings for docker compose
+        from django.conf import settings
+        if options.is_compose:
+            settings.RABBITMQ_HOST = 'zulip-rabbitmq'
+            settings.RABBITMQ_USERNAME = 'zulip',
+            settings.REDIS_HOST = 'zulip-redis'
+            settings.REMOTE_POSTGRES_HOST = 'zulip-postgres'
```

Comment:
So, this isn't going to work, because those changes will be lost as soon as `provision` finishes.  You'll want to write a file inside the Docker container that `zproject/settings.py` (or `zproject/dev_settings.py`) can check for, and if that file exists, set these settings directly.